controllers/search_test.go

```diff
@@ -22,6 +23,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
+func containVolumes(volumes []corev1.Volume, expectedVal string) error {
+	for _, v := range volumes {
+		if v.Name == expectedVal {
+			return nil
+		}
+	}
+	return fmt.Errorf("%s does not exist within the listed volumes env", expectedVal)
+}
```

Comment:
You could pass `t` and validate within this function.

```suggestion
func verifyContainsVolume(t *testing.T, volumes []corev1.Volume, expectedVolume string) {
	for _, v := range volumes {
		if v.Name == expectedVolume {
			return
		}
	}
	t.Errorf("Failed to find to find volume: %s", expectedVolume)
}
```

Reply:
Updated the test function to handle the validation.